src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c

```diff
@@ -108,127 +98,4928 @@ static ColliderCylinderInit D_808E4C2C = {
+        if (eff->type == DORF_EFF_SPARKLE) {
+            gDPPipeSync(POLY_XLU_DISP++);
+            if (flag == 0) {
+                gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 0, 0);
+                gSPDisplayList(POLY_XLU_DISP++, gDorfLightBallMatDL);
+                flag++;
+            }
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, eff->alpha);
+            Matrix_Translate(eff->pos.x, eff->pos.y, eff->pos.z, MTXMODE_NEW);
+            func_800D1FD4(&globalCtx->mf_11DA0);
+            Matrix_Scale(eff->scale, eff->scale, 1.0f, MTXMODE_APPLY);
+            Matrix_RotateZ(eff->unk_3C, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx, "../z_boss_ganon.c", 10932),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, gDorfSquareDL);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon/func_808DF4F0.s")
+    eff = effFirst;
+    flag = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon/func_808DFBD0.s")
+    for (i = 0; i < 150; i++, eff++) {
+        if (eff->type == DORF_EFF_LIGHT_RAY) {
+            gDPPipeSync(POLY_XLU_DISP++);
+            if (flag == 0) {
+                gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 0, 0);
+                gSPDisplayList(POLY_XLU_DISP++, gDorfLightBallMatDL);
+                flag++;
+            }
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, eff->alpha);
+            Matrix_Translate(eff->pos.x, eff->pos.y, eff->pos.z, MTXMODE_NEW);
+            Matrix_RotateY(eff->unk_48, MTXMODE_APPLY);
+            Matrix_RotateX(eff->unk_44, MTXMODE_APPLY);
+            Matrix_RotateZ(eff->unk_3C, MTXMODE_APPLY);
+            Matrix_Scale(eff->scale, eff->scale, eff->unk_38 * eff->scale, MTXMODE_APPLY);
+            Matrix_RotateX(M_PI / 2, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx, "../z_boss_ganon.c", 10971),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, gDorfSquareDL);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon/func_808DFFC8.s")
+    eff = effFirst;
+    flag = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon/func_808E0254.s")
+    for (i = 0; i < 150; i++, eff++) {
+        if (eff->type == DORF_EFF_SHOCK) {
+            if (flag == 0) {
+                gDPPipeSync(POLY_XLU_DISP++);
+                if (eff->unk_2E == DORF_SHOCK_PLAYER_PURPLE) {
+                    // purple
+                    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 100, 0, 200, 255);
+                    gDPSetEnvColor(POLY_XLU_DISP++, 130, 0, 0, 0);
+                } else {
+                    // yellow
+                    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, 255);
+                    gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 0, 0);
```

Comment:
"purple" applies to the primcolor in the first branch, but "yellow" applies to the envcolor in the second branch, what do these colors mean exactly

Reply:
![image](https://user-images.githubusercontent.com/22045047/144076782-14a0eeb2-d151-432b-886c-402981ac8c4a.png)

Reply:
So I took a look at `gDorfShockDL`

how it works is the texture `ovl_Boss_GanonTex_020370` is a "mask" (I8 image), where the bolts are white (255) and the background is black (0)
then from how the combiner is set up, the more white that mask is the more primcolor is used, and the more black the more envcolor
So basically the primcolor is the bolt "inner color" and the envcolor is the "fadeaway color"

you can indeed see on the screenshots,
on the left the purple fades to red
on the right the bolt really is white but fading to yellow

---------------

Anyway I was confused because I thought the comments were about the color set in the dlist commands, not the overall look
May I suggest:
```suggestion
                    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 100, 0, 200, 255);
                    gDPSetEnvColor(POLY_XLU_DISP++, 130, 0, 0, 0);
                } else { // DORF_SHOCK_PLAYER_YELLOW
                    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, 255);
                    gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 0, 0);
```